core/vm/vmcontext/runner.go

```diff
@@ -0,0 +1,120 @@
+package vmcontext
+
+import (
+	"math/big"
+
+	"github.com/celo-org/celo-blockchain/common"
+	"github.com/celo-org/celo-blockchain/core/state"
+	"github.com/celo-org/celo-blockchain/core/types"
+	"github.com/celo-org/celo-blockchain/core/vm"
+)
+
+// VMAddress is the address the VM uses to make internal calls to contracts
+var VMAddress = common.ZeroAddress
+
+// ExtendedChainContext extends ChainContext providing relevant methods
+// for EVMRunner creation
+type ExtendedChainContext interface {
+	vm.ChainContext
+
+	// GetVMConfig returns the node's vm configuration
+	GetVMConfig() *vm.Config
+
+	CurrentHeader() *types.Header
+
+	State() (*state.StateDB, error)
+}
+
+// SystemEVMRunnerBuilder returns a builder factory for a System's EVMRunner
+// DO NOT USE, only for backwards compatibility
+func SystemEVMRunnerBuilder(chain ExtendedChainContext) func(header *types.Header, state vm.StateDB) (vm.EVMRunner, error) {
+	return func(header *types.Header, state vm.StateDB) (vm.EVMRunner, error) {
+		if header == nil {
+			return NewSystemEVMRunnerForCurrentBlock(chain)
+		}
+
+		return NewSystemEVMRunner(chain, header, state), nil
+	}
+}
+
+type evmRunner struct {
+	newEVM func() *vm.EVM
+	state  vm.StateDB
+
+	dontMeterGas bool
+}
+
+// NewSystemEVMRunnerForCurrentBlock creates an EVMRunner where calls are originated by the VMAddress (uses chain's current block)
+func NewSystemEVMRunnerForCurrentBlock(chain ExtendedChainContext) (vm.EVMRunner, error) {
+	return newEVMRunnerForCurrentBlock(chain, VMAddress, common.Big0)
+}
+
+// NewSystemEVMRunner creates an EVMRunner where calls are originated by the VMAddress
+func NewSystemEVMRunner(chain ExtendedChainContext, header *types.Header, state vm.StateDB) vm.EVMRunner {
+	return newEVMRunner(chain, VMAddress, common.Big0, header, state)
```

Comment:
Would it make sense to remove `sender` `gasPrice` as arguments from `newEVMRunnerForCurrentBlock` and `newEVMRunner`?  They are only called with the zero values, and we made them private to indicate they shouldn't be called directly, so I don't see a reason for them to take these arguments unless we envision their getting different values in the future.

Reply:
Right, the reason I thought of this was because on `BoundMethod` you removed the non-system-address varieties of query and execute, so I thought maybe the same rationale would apply here.